docs/design.md

```diff
@@ -0,0 +1,309 @@
+
+#### Attribute grouping
+Attributes can be grouped using JSON objects. Emporous will group the attribute using the schema ID as the root key. This allows for additional filtering by schema, if desired.
+Attribute grouping also adds support for individual signing options (e.g. DSSE envelopes) much like software attestation bundles.
+
+### Dynamic Schema Registration
+
+Attribute schemas can be used at artifact build time to ensure that the attributes provided and gathered are formatted as expected. JSON Schema would be published just like any other
+content, which makes them discoverable through the attributes API.
+
+#### Static Schema Validation
+In this instance, the registry would still validate each manifest against the OCI image spec and any additional schema validation would be a registry specific implementation. The goal is
+to change the OCI image spec to require registries to be attribute-aware, not attribute schema-aware. 
+
+### Aggregates
+
+The attribute query endpoint can be used to automate building applications from existing artifacts. The endpoint will resolve an attribute query to an index manifest of matching descriptors that can be used for artifact publishing.
+
+To enable efficiency in this workflow, it would be best practice to only publish one component or file per artifact. 
+However, if an artifact is published with more than one file, the Emporous client will allow attribute filtering on individual artifacts.
+
+### Search Domains
+
+A content query can be sent to multiple OCI registries. A user can specify which registries to query and prioritize registry results.
+If an artifact or aggregate references a link that cannot be resolved by any registry in the search domain, a field called `registryHint` will be
+used to add the registry with the required content to a discovered zone.
+
+### Reference Types/Links
+
+Emporous artifacts must be able to link to other artifacts.  Information on application dependencies is just one of many examples of this. 
+Below, we examine a few techniques for achieving artifact linking.
+
+#### OCI Reference types
+
+Referrers are required to employ a tag schema to specify associations in the `subject` field, as per the [OCI Referrers API spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers). An index manifest answer is produced in response to an API request for the referrer.
+When developing supporting artifacts for an artifact that would only ever refer to one artifact as the parent, this kind of approach may be used.
+
+##### Example Use Case
+
+```mermaid
+graph LR;
+A[Signature]-->B[Artifact];
+C[Attestation] --> B
+D[Other] --> B
+```
+
+#### Artifact Link
+
+Links are an Emporous specific type that differs from referrers in that they define cross-repo relationships and many-to-many relationships between artifacts.
```

Comment:
cross-repo is limiting. I prefer the term cross-namespace, which encompasses cross-registry. 
```suggestion
Links are an Emporous specific type that differs from referrers in that they define cross-namespace relationships and many-to-many relationships between artifacts.
```